src/error.rs

```diff
@@ -52,3 +54,16 @@ impl From<i32> for Error {
         }
     }
 }
+
+impl <'a> std::fmt::Display for Error <'_> {
+    //TODO: Add documentation 
+    pub fn fmt(&self, f: &mut std::fmt::Formatter <'_> ) -> std::fmt::Result {
+        match *self {
+            Error::NoError(str: &str) => wirteln!(io::stderr(), "{}", str),
```

Comment:
This needs to be removed. There is no need to handle a case where there is not an error. IIRC this is also only used  in `main.rs`?